source/Octopus.Tentacle/Communications/HalibutInitializer.cs

```diff
@@ -134,5 +135,19 @@ IPEndPoint GetEndPointToListenOn()
         public void Stop()
         {
         }
+        
+        public void Dispose()
```

Comment:
🤔 This struck me as odd that we have to do this, as Autofac naturally disposes of dependencies given to a class.

I tested this, and this is indeed true. So why did we need to do this for Halibut?

Turns out it's the difference between `IDisposable` and `IAsyncDisposable`. The former will get disposed by Autofac, and the latter will not 😑.

Since we recently removed the `IDisposable` interface from Halibut, this is why it has stopped being disposed of by Autofac.

Knowing all this, it made me wonder:
- What we did here will work. But would it be better or worse to put back the `HalibutRuntime.Dispose` method as an alternative to solving this problem?
- Also, we end up double disposing `log`, as Autofac also disposes it (I tested it to make sure).